### PR TITLE
Docs: Add towncrier

### LINKED ISSUE #92: Use towncrier for changelog generation
I want to establish the usage of https://github.com/twisted/towncrier paired with GitHub Actions to enable me to spend less time writing the changelog.

At the moment this process can take hours and I want to save that time and use it to develop meaningful features.

The new release process would be:

1. Trigger a manual run of the action that creates a PR with the generated changelog.
2. Accept the opened PR
3. GH Actions detects this and triggers a run of the action that creates the release.
4. I manually increase the version to the one that is specified as the next version.

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #92 

Successful test for the changelog creation:

- https://github.com/SchoolGuy/koan/actions/runs/3882382407
- https://github.com/SchoolGuy/koan/pull/6

Successful test for the automatic release creation:

- https://github.com/SchoolGuy/koan/actions/runs/3891236556
- https://github.com/SchoolGuy/koan/releases/tag/v3.0.6

### CODE REVIEW

**Comment by SchoolGuy** [comment]
We will not touch the PyPi Workflow. I will fix this in another PR.

File: towncrier.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[tool.towncrier]
+directory = "changelog.d"
+filename = "changelog/v{version}.md"
+package = "koan"
+start_string = "<!-- towncrier release notes start -->\n"
+underlines = ["", "", ""]
+template = "changelog.d/changelog_template.jinja"
+title_format = "# Koan [{version}](https://github.com/SchoolGuy/koan/tree/v{version}) - {project_date}"
+issue_format = "[#{issue}](https://github.com/SchoolGuy/koan/issues/{issue})"
```
Comment (tiltingpenguin) [inline towncrier.toml:9]: This is referencing the testing fork

**Review by tiltingpenguin — CHANGES_REQUESTED** [review]
Just found one small thing you forgot

File: towncrier.toml (hunk 1 above)
Comment (SchoolGuy) [reply towncrier.toml:9]: Fixed

**Review by tiltingpenguin — APPROVED** [review]
LGTM

**Pull request merged**